### ISSUE TITLE
Can't create new tickets when having filters

### ISSUE THREAD

**Issue by jacobshilitz** [osTicket/osTicket#6131]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [X] Do you have any errors in the PHP error log, or javascript console?
* [ ] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [X] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description
Hello, 

Found this problem, sorry, I have no time to test on new environment and read the forums, but I just want to report back, as looking on the code there is no way this is correct.

After pull https://github.com/osTicket/osTicket/pull/5947 where a new feature was added to "Log Events For Filters",
I got this error: `Fatal error: Call to a member function logEvent() on boolean`.

The line above is the **troublemaker**, it happens because `$postCreate` is a bool it cannot call `logEvent()`. 

https://github.com/osTicket/osTicket/blob/d22cfaccdcc63bff9a472271fe479b7af699737c/include/class.ticket.php#L4036
I understand that `logEvent()` has to be called on a ticket instance, I didn't have time to play with it, to check how to access the ticket instance at the point of the filter, I suspect that the ticket is created only after all filters are completed.

### Logs 
```
[Mon Apr 11 12:59:56.952647 2022] [php7:error] [pid 29172] [client ip:53522] PHP Fatal error:  Uncaught Error: Call to a member function logEvent() on boolean in /var/www/html/include/class.ticket.php:4063\nStack trace:\n#0 /var/www/html/include/class.ticket.php(4483): Ticket::filterTicketData('staff', Array, Array, Object(User), true)\n#1 /var/www/html/include/class.ticket.php(4682): Ticket::create(Array, Array, 'staff', false)\n#2 /var/www/html/scp/tickets.php(432): Ticket::open(Array, Array)\n#3 {main}\n  thrown in /var/www/html/include/class.ticket.php on line 4063, referer: https://doamin/scp/tickets.php?a=open
```

### Steps to Reproduce

1. create a new filter 
2. create a new ticket that would have trigger the filter

**Expected behavior:** 
Ticket creation with no errors
**Actual behavior:** 
![image](https://user-images.githubusercontent.com/26477664/162806918-8231ef56-f294-4356-b773-b30d11de5fc8.png)
### Versions

osTicket Version | v1.15.6 (1a64b84) —  Up to date
Web Server Software | Apache/2.4.29 (Ubuntu)
MySQL Version | 5.7.33
PHP Version | 7.2.24-0ubuntu0.18.04.11

**Comment by ntozier** [comment]
You might want to upgrade to current since you are running an old version.  1.16.x is current.

**Comment by jacobshilitz** [comment]
I'm not ready for PHP 8, but there is no fix for this bug in v1.16
